### PR TITLE
Complete connector separation of SharePoint Server from SharePoint Online

### PR DESCRIPTION
**Pull request by akanshi-elastic**
## Closes https://github.com/elastic/connectors-python/issues/1008 ###

This PR contains:
1.file renaming from sharepoint.py-> sharepoint_server.py
2.Remove the code of sharepoint online from the file
3.Update readme changes for sharepoint server only

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [x] Considered corresponding documentation changes

### CODE REVIEW

File: connectors/sources/sharepoint_server.py [hunk 1]
```diff
@@ -414,51 +333,22 @@ async def _api_call(self, url_name, url="", **url_kwargs):
...
-                    retry=retry, exception=exception
-                )
-                await self.close_session()
-                logger.debug(f"Will retry in {retry_seconds} seconds")
-                await self._sleeps.sleep(retry_seconds)
-
             except Exception as exception:
+                if isinstance(
+                    exception,
+                    ServerDisconnectedError,
```
Comment (artem-shelkovnikov) [inline connectors/sources/sharepoint_server.py:349]: No retrying after logic here?

File: connectors/sources/sharepoint_server.py [hunk 2]
```diff
@@ -699,92 +605,51 @@ def get_default_configuration(cls):
...
-                    {"label": "SharePoint Server", "value": SHAREPOINT_SERVER},
-                ],
-                "order": 1,
-                "type": "str",
-                "value": SHAREPOINT_SERVER,
-            },
             "username": {
-                "depends_on": [{"field": "data_source", "value": SHAREPOINT_SERVER}],
                 "label": "SharePoint Server username",
                 "order": 2,
```
Comment (artem-shelkovnikov) [inline connectors/sources/sharepoint_server.py:610]: Order should start from 1 - so you need to decrement all others below by 1 too

File: connectors/sources/sharepoint_server.py (hunk 1 above)
Comment (akanshi-elastic) [reply connectors/sources/sharepoint_server.py:349]: Yes retry is there on line 352 we have just wrap the complete retry in a separate method.

**Comment by artem-shelkovnikov** [comment]
@akanshi-elastic I experienced this problem during testing:

image

Seems like there's a problem with configuration, can you address it?

**Review by artem-shelkovnikov — APPROVED** [review]

**Pull request merged**